plenario/api/validator.py

```diff
@@ -0,0 +1,173 @@
+    data_type = fields.Str(default='json', validate=OneOf(valid_formats))
+    location_geom__within = fields.Str(default=None, dump_to='geom')
+    obs_date__ge = fields.Date(default=datetime.now() - timedelta(days=90))
+    obs_date__le = fields.Date(default=datetime.now())
+    offset = fields.Integer(default=0, validate=Range(0))
+    resolution = fields.Integer(default=500, validate=Range(0))
+    shape = fields.Str(default=None, validate=OneOf(ShapeMetadata.index()))
+
+
+class DatasetRequiredValidator(Validator):
+    """Some endpoints, like /detail-aggregate, should not be allowed to recieve
+    requests that do not specify a 'dataset_name' in the query string."""
+
+    dataset_name = fields.Str(default=None, validate=OneOf(MetaTable.index()), dump_to='dataset', required=True)
+
+
+class NoDefaultDatesValidator(Validator):
+    """Some endpoints, specifically /datasets, will not return results with
+    the original default dates (because the time window is so small)."""
+
+    obs_date__ge = fields.Date(default=None)
+    obs_date__le = fields.Date(default=None)
+
+
+# ValidatorResult
+# ===============
+# Since much of the old code, formatting results into json, expected data as
+# attributes to the old ParamValidator object, it was convenient to use a named
+# tuple here that made a validator result compatible with response code.
+
+ValidatorResult = namedtuple('ValidatorResult', 'data errors warnings')
+
+
+# converters
+# ==========
+# Callables which are used to convert request arguments to their correct types.
+
+converters = {
+    'buffer': int,
+    'dataset': lambda x: MetaTable.get_by_dataset_name(x),
+    'dataset_name__in': lambda x: x.split(','),
+    'date__time_of_day_ge': int,
+    'date__time_of_day_le': int,
+    'obs_date__ge': lambda x: parser.parse(x).date(),
+    'obs_date__le': lambda x: parser.parse(x).date(),
+    'offset': int,
+    'resolution': int,
+    'geom': lambda x: make_fragment_str(extract_first_geometry_fragment(x)),
+    'shape': lambda x: get_shape_table(x)
+}
+
+
+def get_shape_table(shtable_name):
+    """A simple helper method to get a table that corresponds to a ShapeMetadata
+    record. This exists because while MetaTable.point_table exists, ShapeMetadata
+    doesn't have an analogous function.
+
+    :param shtable_name: name of shape table
+
+    :returns: table instance corresponding to ShapeMetadata record"""
+
+    return session.query(ShapeMetadata)\
+        .filter(ShapeMetadata.dataset_name == shtable_name)\
+        .first()\
+        .shape_table
+
+
+def convert(request_args):
+    """Convert a dictionary of arguments from strings to their types. How the
+    values are converted are specified by the converters dictionary defined
+    above.
+
+    :param request_args: dictionary of request arguments
+
+    :returns: converted dictionary"""
+
+    for key, value in request_args.items():
+        try:
+            request_args[key] = converters[key](value)
+        except:
```

Comment:
This seems dangerous. Any exception triggers a DB rollback, but it looks like the loop happily goes on from there. Do we not want to fail completely?